site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,281 @@
+
+Speculation rules in subframes are not acted upon, and speculation rules in prerendered pages are only acted upon once the page is activated.
+
+Below is a simple example of adding a `prerender` speculation rule:
+
+```js
+if (HTMLScriptElement.supports &&
+    HTMLScriptElement.supports('speculationrules')) {
+  const specScript = document.createElement('script');
+  specScript.type = 'speculationrules';
+  specRules = {
+    prerender: [
+      {
+        source: 'list',
+        urls: ['/next.html'],
+      },
+    ],
+  };
+  specScript.textContent = JSON.stringify(specRules);
+  console.log('added speculation rules to: next.html');
+  document.body.append(specScript);
+}
+```
+
+You can view a demo of Speculation Rules API prerendering on [this demo page](https://prerender-demos.glitch.me/).
+
+## Detecting and disabling prerendering
+
+Prerender is usually a positive experience for users as it allows fast page rendering—often instant. This benefits both the user, and the site owner, since prerendered pages allow a better user experience that may be difficult to achieve otherwise.
+
+However, there may be instances when you do not wish prerendering of pages to happen, for example when pages change state—either based on the initial request, or based on JavaScript executing on the page.
+
+### Enabling and disabling prerender in Chrome
+
+Prerender is only enabled for those Chrome users with the "Preload pages for faster browsing and searching" setting in `chrome://settings/cookies/`. Additionally, prerender is also disabled on low-memory devices, or if the operating system is in data-saver or battery-saver modes.
+
+### Detecting and disabling prerender server-side
+
+Prerendered pages will be sent with the following HTTP header:
+
+```http
+Sec-Purpose: prefetch;prerender
+```
+
+Servers can respond based on this header, to log prerender requests, return different content, or prevent the prerender from happening. If a non-success response code is returned (that is, not a 200 or a 304), then the page will not be prerendered.
+
+If you absolutely do not want a particular page to be prerendered, this is the best way to ensure it won't happen.
+
+### Detecting prerender in JavaScript
+
+The `document.prerendering` API will return `true` while the page is prerendering. This can be used by pages to prevent—or delay—certain activities during the prerender until the page is actually activated.
+
+Once a prerendered document is activated, `PerformanceNavigationTiming`'s `activationStart` will also be set to a non-zero time representing the time between when the prerender was started and the document was actually activated.
+
+You can have a function to check for _prendering_ and _prerendered_ pages like the following:
+
+```js
+function pagePrerendered() {
```

Comment:
FYI, I discovered a better way to do this:

```js
function pagePrerendered() {
  return (
    document.prerendering ||
    self.performance?.getEntriesByType?.('navigation')[0]?.activationStart > 0
  );
}
```

Reply:
Updated